zebra-rpc/src/config.rs

```diff
@@ -55,15 +55,22 @@ pub struct Config {
     pub debug_force_finished_sync: bool,
 }
 
+// This impl isn't derivable because it depends on features.
+#[allow(clippy::derivable_impls)]
 impl Default for Config {
     fn default() -> Self {
         Self {
             // Disable RPCs by default.
             listen_addr: None,
 
             // Use a single thread, so we can detect RPC port conflicts.
+            #[cfg(not(feature = "getblocktemplate-rpcs"))]
             parallel_cpu_threads: 1,
 
+            // Use multiple threads, because we pause requests during getblocktemplate long polling
```

Comment:
It probably does whatever the tokio executor does, because that's what it uses internally to handle futures on its threads.

So it will block the thread if the future does a blocking task, or uses a lot of CPU, or doesn't yield. But otherwise it will be ok.